anta/result_manager/__init__.py

```diff
@@ -109,106 +123,95 @@ def _update_status(self, test_status: TestStatus) -> None:
         elif self.status == "success" and test_status == "failure":
             self.status = "failure"
 
-    def add_test_result(self, entry: TestResult) -> None:
-        """Add a result to the list.
+    def add(self, result: TestResult) -> None:
+        """Add a result to the report.
 
         Args:
         ----
-            entry (TestResult): TestResult data to add to the report
+            test: TestResult data to add to the report.
 
         """
-        logger.debug(entry)
-        self._result_entries.append(entry)
-        self._update_status(entry.result)
-
-    def add_test_results(self, entries: list[TestResult]) -> None:
-        """Add a list of results to the list.
-
-        Args:
-        ----
-            entries (list[TestResult]): List of TestResult data to add to the report
-
-        """
-        for e in entries:
-            self.add_test_result(e)
+        logger.debug(result)
+        self._result_entries.append(result)
+        self._update_status(result.result)
 
     def get_status(self, *, ignore_error: bool = False) -> str:
         """Return the current status including error_status if ignore_error is False."""
         return "error" if self.error_status and not ignore_error else self.status
 
-    def get_results(self) -> list[TestResult]:
-        """Expose list of all test results in different format.
+    def filter(self, *, hide_success: bool = False, hide_failure: bool = False, hide_error: bool = False) -> ResultManager:
+        """Get a filtered ResultManager based on test status.
 
-        Returns
-        -------
-            any: List of results.
-
-        """
-        return self._result_entries
-
-    def get_json_results(self) -> str:
-        """Expose list of all test results in JSON.
+        Args:
+        ----
+            hide_success: Hide tests with status `success` if True.
+            hide_failure: Hide tests with status `failure` if True.
+            hide_error: Hide tests with status `error` if True.
 
         Returns
         -------
-            str: JSON dumps of the list of results
-
+            A filtered `ResultManager`.
         """
-        result = [result.model_dump() for result in self._result_entries]
-        return json.dumps(result, indent=4)
+        manager = ResultManager()
+        manager.results = [
+            test
+            for test in self._result_entries
+            if (not (hide_success and test.result == "success") and not (hide_failure and test.result == "failure") and not (hide_error and test.result == "error"))
+        ]
+        return manager
 
-    def get_result_by_test(self, test_name: str) -> list[TestResult]:
-        """Get list of test result for a given test.
+    def filter_by_test(self, tests: list[str]) -> ResultManager:
```

Comment:
so here we actually return a different object  than the initial one
I am not saying this is bad just that it could be confusing in usage if trying to chain the functions